src/main/java/org/apache/sling/api/uri/SlingUri.java

```diff
@@ -0,0 +1,233 @@
+ * 
+ * @since 1.0.0 (Sling API Bundle 2.23.0)
+ */
+@ProviderType
+public interface SlingUri extends RequestPathInfo {
+
+    /**
+     * Returns the @{link URI}.
+     * 
+     * @return the URI
+     */
+    @NotNull
+    URI toUri();
+
+    /**
+     * Returns the URI as String.
+     * 
+     * @return the URI string
+     */
+    @NotNull
+    String toString();
+
+    /**
+     * Returns the scheme.
+     * 
+     * @return the scheme or null if not set
+     */
+    @Nullable
+    String getScheme();
+
+    /**
+     * Returns the user info.
+     * 
+     * @return the user info of the SlingUri or null if not set
+     */
+    @Nullable
+    String getUserInfo();
+
+    /**
+     * Returns the host.
+     * 
+     * @return returns the host of the SlingUri or null if not set
+     */
+    @Nullable
+    String getHost();
+
+    /**
+     * Returns the port.
+     * 
+     * @return returns the port of the SlingUri or -1 if not set
+     */
+    int getPort();
+
+    /**
+     * Returns the resource path.
+     * 
+     * @return returns the resource path or null if the URI does not contain a path.
+     */
+    @Override
+    @Nullable
+    String getResourcePath();
+
+    /**
+     * Returns the selector string.
+     * 
+     * @return returns the selector string or null if the URI does not contain selector(s)
+     */
+    @Override
+    @Nullable
+    String getSelectorString();
+
+    /**
+     * Returns the selectors array.
+     * 
+     * @return the selectors array (empty if the URI does not contain selector(s))
+     */
+    @Override
+    String[] getSelectors();
+
+    /**
+     * Returns the extension.
+     * 
+     * @return the extension or null if the URI does not contain an extension
+     */
+    @Override
+    @Nullable
+    String getExtension();
+
+    /**
+     * Returns the path parameters.
+     * 
+     * @return the path parameters or an empty Map if the URI does not contain any
+     */
+    Map<String, String> getPathParameters();
+
+    /**
+     * Returns the suffix part of the URI
+     * 
+     * @return the suffix string or null if the URI does not contain a suffix
+     */
+    @Override
+    @Nullable
+    String getSuffix();
+
+    /**
+     * Returns the joint path of resource path, selectors, extension and suffix.
+     * 
+     * @return the path or null if no path is set
+     */
+    @Nullable
+    String getPath();
+
+    /**
+     * Returns the query.
+     * 
+     * @return the query part of the URI or null if the URI does not contain a query
+     */
+    @Nullable
+    String getQuery();
+
+    /**
+     * Returns the fragment.
+     * 
+     * @return the fragment or null if the URI does not contain a fragment
+     */
+    @Nullable
+    String getFragment();
+
+    /**
+     * Returns the scheme-specific part of the URI, compare with Javadoc of class
+     * <a href="https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/net/URI.html">URI</a>.
```

Comment:
as Sling API is compliant with Java 8, please link Java 8 javadocs. You can use the link tag to let javadoc generate the link from the FQCN automatically.